streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -316,20 +313,22 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             classifyTasksWithStateUpdater(activeTasksToCreate, standbyTasksToCreate, tasksToRecycle, tasksToCloseClean);
         }
 
-        tasks.addPendingActiveTasks(pendingTasksToCreate(activeTasksToCreate));
-        tasks.addPendingStandbyTasks(pendingTasksToCreate(standbyTasksToCreate));
+        tasks.purgePendingTasksToCreate();
+        tasks.addPendingActiveTasksToCreate(pendingTasksToCreate(activeTasksToCreate));
+        tasks.addPendingStandbyTasksToCreate(pendingTasksToCreate(standbyTasksToCreate));
```

Comment:
I was actually proposing to move this block before line 306.

Reply:
ack.